momento-sdk/src/test/java/momento/sdk/responses/storage/GetResponseTest.java

```diff
@@ -1,64 +1,67 @@
 package momento.sdk.responses.storage;
 
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
 import io.grpc.Status;
+import momento.sdk.exceptions.ClientSdkException;
 import momento.sdk.exceptions.StoreNotFoundException;
 import momento.sdk.internal.MomentoGrpcErrorDetails;
 import momento.sdk.internal.MomentoTransportErrorDetails;
 import org.junit.jupiter.api.Test;
 
 public class GetResponseTest {
   @Test
-  public void testGetResponseSuccessWorksOnTheRightType() {
-    GetResponse.Success response = GetResponse.Success.of(new byte[] {0, 1, 2, 3});
-    assert response.value().get().getByteArray().get().length == 4;
-    assert !response.value().get().getString().isPresent();
-    assert !response.value().get().getLong().isPresent();
-    assert !response.value().get().getDouble().isPresent();
+  public void testGetResponseFoundWorksOnTheRightType() {
+    GetResponse.Found response = GetResponse.Found.of(new byte[] {0, 1, 2, 3});
+    assert response.value().getByteArray().get().length == 4;
+    assertThrows(ClientSdkException.class, response.value().getString()::orElseThrow);
+    assertThrows(ClientSdkException.class, response.value().getLong()::orElseThrow);
+    assertThrows(ClientSdkException.class, response.value().getDouble()::orElseThrow);
```

Comment:
I see that in this found situation, even for the use case of orElseThrow, the presence of GetResponseFound means that value() line noise is required to get at the value.
I think when a user has gotten to a point in the type system where a response is known to have been found, they should be able to call getString().orElseThrow() directly.
Further, have you considered making life easier for the throwy person and using `get*` and `as*` methods instead? `getString()` could throw, while `asString()` could return your optionaly type.
This would make these lines instead look like
```suggestion
    // When found, a value is a StoreValue.
    StoreValue response = StoreValue.of(new byte[] {0, 1, 2, 3});
    assert response.asByteArray().get().length == 4; // could use getByteArray here instead
    assertThrows(ClientSdkException.class, response::getString);
    assertThrows(ClientSdkException.class, response::getLong);
    assertThrows(ClientSdkException.class, response::getDouble);
```

I love the exhaustive pattern as you have it, minus perhaps the GetResponseFound indirection. I think the throwy path could stand to be trimmed still further to give a user exactly what they expect or else an exception straight away. Happy to chat more about it if you like!

Reply:
> I think when a user has gotten to a point in the type system where a response is known to have been found, they should be able to call getString().orElseThrow() directly.

Agreed and will consider this.

> Further, have you considered making life easier for the throwy person and using get* and as* methods instead? getString() could throw, while asString() could return your optionaly type.

Yes I had this in a previous commit. We agreed internally to keep it slim for now and to add a parallel track of throwy methods per customer feedback.

Reply:
+1 to keeping slim for now. We can 'throwy' versions of things in the future if there is interest.